// pliny/chunk-4LUJJPSD.js
import {
  formatDate
} from "./chunk-T2LUFWMO.js";

// src/search/KBarPortal.tsx
import { useState, useEffect } from "react";
import Router from "next/router.js";
import {
  KBarPortal,
  KBarSearch,
  KBarAnimator,
  KBarPositioner,
  KBarResults,
  useMatches,
  useRegisterActions,
  useKBar
} from "kbar";
import { jsx, jsxs } from "react/jsx-runtime";
"use client";
var init = false;
var Portal = ({ searchDocumentsPath }) => {
  const [searchActions, setSearchActions] = useState([]);
  const { query } = useKBar();
  useEffect(() => {
    if (!init) {
      init = true;
      query.toggle();
    }
  }, []);
  useEffect(() => {
    const mapPosts = (posts) => {
      const actions = [];
      for (const post of posts) {
        actions.push({
          id: post.path,
          name: post.title,
          keywords: (post == null ? void 0 : post.summary) || "",
          section: "Content",
          subtitle: formatDate(post.date, "en-US"),
          perform: () => Router.push("/" + post.path)
        });
      }
      return actions;
    };
    async function fetchData() {
      const res = await fetch(searchDocumentsPath);
      const json = await res.json();
      const actions = mapPosts(json);
      setSearchActions(actions);
    }
    fetchData();
  }, [searchDocumentsPath]);
  useRegisterActions(searchActions, [searchActions]);
  return /* @__PURE__ */ jsx(KBarPortal, {
    children: /* @__PURE__ */ jsx(KBarPositioner, {
      className: "bg-gray-300/50 p-4 backdrop-blur backdrop-filter dark:bg-black/50",
      children: /* @__PURE__ */ jsx(KBarAnimator, {
        className: "w-full max-w-xl",
        children: /* @__PURE__ */ jsxs("div", {
          className: "overflow-hidden rounded-2xl border border-gray-100 bg-gray-50 dark:border-gray-800 dark:bg-gray-900",
          children: [
            /* @__PURE__ */ jsxs("div", {
              className: "flex items-center space-x-4 p-4",
              children: [
                /* @__PURE__ */ jsx("span", {
                  className: "block w-5",
                  children: /* @__PURE__ */ jsx("svg", {
                    className: "text-gray-400 dark:text-gray-300",
                    xmlns: "http://www.w3.org/2000/svg",
                    fill: "none",
                    viewBox: "0 0 24 24",
                    stroke: "currentColor",
                    children: /* @__PURE__ */ jsx("path", {
                      strokeLinecap: "round",
                      strokeLinejoin: "round",
                      strokeWidth: 2,
                      d: "M21 21l-6-6m2-5a7 7 0 11-14 0 7 7 0 0114 0z"
                    })
                  })
                }),
                /* @__PURE__ */ jsx(KBarSearch, {
                  className: "h-8 w-full bg-transparent text-slate-600 placeholder-slate-400 focus:outline-none dark:text-slate-200 dark:placeholder-slate-500"
                }),
                /* @__PURE__ */ jsx("span", {
                  className: "inline-block whitespace-nowrap rounded border border-slate-400/70 px-1.5 align-middle font-medium leading-4 tracking-wide text-slate-500 [font-size:10px] dark:border-slate-600 dark:text-slate-400",
                  children: "ESC"
                })
              ]
            }),
            /* @__PURE__ */ jsx(RenderResults, {})
          ]
        })
      })
    })
  });
};
var RenderResults = () => {
  const { results } = useMatches();
  if (results.length) {
    return /* @__PURE__ */ jsx(KBarResults, {
      items: results,
      onRender: ({ item, active }) => /* @__PURE__ */ jsx("div", {
        children: typeof item === "string" ? /* @__PURE__ */ jsx("div", {
          className: "pt-3",
          children: /* @__PURE__ */ jsx("div", {
            className: "text-primary-600 block border-t border-gray-100 px-4 pt-6 pb-2 text-xs font-semibold uppercase dark:border-gray-800",
            children: item
          })
        }) : /* @__PURE__ */ jsxs("div", {
          className: `block cursor-pointer px-4 py-2 text-gray-600 dark:text-gray-200 ${active ? "bg-primary-600" : "bg-transparent"}`,
          children: [
            item.subtitle && /* @__PURE__ */ jsx("div", {
              className: `${active ? "text-gray-200" : "text-gray-400 dark:text-gray-500"} text-xs`,
              children: item.subtitle
            }),
            /* @__PURE__ */ jsx("div", {
              children: item.name
            })
          ]
        })
      })
    });
  } else {
    return /* @__PURE__ */ jsx("div", {
      className: "block border-t border-gray-100 px-4 py-8 text-center text-gray-400 dark:border-gray-800 dark:text-gray-600",
      children: "No results for your search..."
    });
  }
};

export {
  Portal
};
