// pliny/chunk-G34OK7QT.js
import { GA } from './chunk-AYXXGBZL.js';
import { Plausible } from './chunk-FY3LR7VV.js';
import { Posthog } from './chunk-XDM7W6R4.js';
import { SimpleAnalytics } from './chunk-4T7S6EVS.js';
import { Umami } from './chunk-FSUSLCEY.js';
import { jsxs, Fragment, jsx } from 'react/jsx-runtime';

var isProduction = process.env.NODE_ENV === "production";
var Analytics = ({ analyticsConfig }) => {
  return /* @__PURE__ */ jsxs(Fragment, {
    children: [
      isProduction && analyticsConfig.plausibleDataDomain && /* @__PURE__ */ jsx(Plausible, {
        plausibleDataDomain: analyticsConfig.plausibleDataDomain
      }),
      isProduction && analyticsConfig.simpleAnalytics && /* @__PURE__ */ jsx(SimpleAnalytics, {}),
      isProduction && analyticsConfig.posthogProjectApiKey && /* @__PURE__ */ jsx(Posthog, {
        posthogProjectApiKey: analyticsConfig.posthogProjectApiKey
      }),
      isProduction && analyticsConfig.umamiWebsiteId && /* @__PURE__ */ jsx(Umami, {
        umamiWebsiteId: analyticsConfig.umamiWebsiteId
      }),
      isProduction && analyticsConfig.googleAnalyticsId && /* @__PURE__ */ jsx(GA, {
        googleAnalyticsId: analyticsConfig.googleAnalyticsId
      })
    ]
  });
};

export { Analytics };
